reflex/experimental/vars/base.py

```diff
@@ -28,6 +29,10 @@ class ImmutableVar(Var):
     # Extra metadata associated with the Var
     _var_data: Optional[VarData] = dataclasses.field(default=None)
 
+    def __post_init__(self):
+        """Post-initialization."""
+        _global_vars[hash(self)] = self
```

Comment:
my initial thought was to store the reference only in `__format__`, to avoid keeping unused references to Vars that never actually get formatted. Saving refs to all Vars could cause memory use issues where Vars go out of scope, but cannot be garbage collected because they are in this global map.